qa/snapshot-repository-downgrade/src/test/java/org/elasticsearch/upgrades/ClusterDowngradeIT.java

```diff
@@ -0,0 +1,220 @@
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.snapshots.RestoreInfo;
+import org.elasticsearch.test.rest.ESRestTestCase;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+
+/**
+ * Downgrade tests that verify that a snapshot repository is not getting corrupted and continues to function properly during cluster
+ * downgrades. Concretely this test suit is simulating the following scenario:
+ * <ul>
+ *     <li>Start from a cluster in an old version: {@link TestStep#OLD}</li>
+ *     <li>Upgrade the cluster to the current version: {@link TestStep#UPGRADED}</li>
+ *     <li>Downgrade the cluster back to the old version: {@link TestStep#DOWNGRADED}</li>
+ *     <li>Once again upgrade the cluster to the current version: {@link TestStep#RE_UPGRADED}</li>
+ * </ul>
+ * TODO: Add two more steps: delete all old version snapshots from the repository, then downgrade again and verify that the repository
+ *       is not being corrupted. This requires first merging the logic for reading the min_version field in RepositoryData back to 7.6.
+ */
+public class ClusterDowngradeIT extends ESRestTestCase {
+
+    protected enum TestStep {
+        OLD,
+        UPGRADED,
+        DOWNGRADED,
+        RE_UPGRADED;
+
+        public static TestStep parse(String value) {
+            switch (value) {
+                case "old_cluster":
+                    return OLD;
+                case "upgraded_cluster":
+                    return UPGRADED;
+                case "downgraded_cluster":
+                    return DOWNGRADED;
+                case "re_upgraded_cluster":
+                    return RE_UPGRADED;
+                default:
+                    throw new AssertionError("unknown test step: " + value);
+            }
+        }
+    }
+
+    protected static final TestStep TEST_STEP = TestStep.parse(System.getProperty("tests.rest.suite"));
+
+    @Override
+    protected boolean preserveIndicesUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected boolean preserveSnapshotsUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected boolean preserveReposUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected boolean preserveTemplatesUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected boolean preserveClusterSettings() {
+        return true;
+    }
+
+    @Override
+    protected boolean preserveRollupJobsUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected boolean preserveILMPoliciesUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected boolean preserveSLMPoliciesUponCompletion() {
+        return true;
```

Comment:
do we need this?

Reply:
Nah I just blindly copy and pasted these, removed all but the repo and snapshot preserving now :)